version_updates.md

```diff
@@ -0,0 +1,130 @@
+# OpenStackVersion
+
+## Overview
+
+This document describes how to use the OpenStackVersion
+CR to apply a minor updates to a deployed OpenStack
+installation (including both Controlplane and Dataplane
+Services)
+
+## Requirements
+
+ * Use of OLM (Operator Lifecycle Manager) is assumed
+ * The CSV (ClusterServiceVersion) for the openstack-operator
+   is used as the primary resource for versioning and the
+   associated ENV variables for OpenStack service containers.
+   These are typically prefixed with RELATED\_IMAGE...
+   The CSV defines the version.
+ * OpenStack operators are always upgraded first via OLM. This
+   can be either via manual or automatic. Upgrading operators
+   will trigger OpenStackVersion to provide access to a new
+   available version.
+ * And ENV called OPENSTACK\_VERSION must be set in the CSV
+   for openstack-operator.  This controls the 'availableVersion'
+   of OpenStackVersion.
+   
+
+## General Update Workflow
+
+OpenStack is installed by following the normal installation procedure. When
+an OpenStackControlplane resource is created in any given namespace an
+OpenStackVersion resource will be immediately created when the OpenStackControlPlane
+is reconciled (1 OpenStackVersion per namespace).
+
```

Comment:
For Cinder volume I think at this point the `cinderVolumeCustomImages` section of the `OpenStackVersion` should be automatically populated if the `containerImage` has been configured for a cinder backend.

That way the `OpenStackVersion` and `OpenStackControlplane` are in sync.

Don't know if it's worth doing it for other services. I think it may be too much.

Reply:
I added a section to describe the way it works now: if the backend does not require a custom container the default cinder
volume container is used automatically. This means if you are only using
backends that require the default cinder volume container no
edits to the OpenStackVersion resource are needed